File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        
        ## convert nums array into a set
        NumSet = set(nums)
        longest = 0
        
        for n in nums:
            ## check if its the starting of a sequence
            if(n-1) not in NumSet:
                length = 0
                ##check current numer in numset or not
                while(n+length) in NumSet:
                    length+=1
                    
                longest  = max(length , longest)
                
        return longest      
        

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
// time:o(n)
// space:o(n)
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        
        unordered_set<int> NumSet(nums.begin() , nums.end());
        int longest = 0;
            
        for(int n : nums){
            if(NumSet.find(n-1) == NumSet.end()){
                int length = 0;
                    
                while(NumSet.find(n+length) != NumSet.end()){
                    length++;
                }    
                longest = max(length , longest);
            }
        } 
        return longest;
        
    }
};